packages/core/bootstrap/src/lib/modules/selector.ts

```diff
@@ -113,6 +115,96 @@ const selectEndpoint = <C extends Config>(
   return apiEndpoint
 }
 
+/**
+ *
+ * @param request the request coming in to the External Adapter
+ * @param downstreamConfig configuration for the downstream adapter (composite)
+ * @param downstreamEndpoints endpoints for the downstream adapter (composite)
+ * @param upstreamEndpointsGroups endpoint groups for the upstream adapter (source)
+ *
+ * The following data structure is used:
+ *
+ * [
+ *
+ *    input property name of upstream adapter name,
+ *
+ *    upstream Endpoints mapped to UPPERCASE adapter names,
+ *
+ *    the endpoint name to select upstream
+ *
+ * ]
+ *
+ * @param upstreamConfig configuration for the upstream adapter (source)
+ * @param ignoreRequired whether to maintain the required status of input parameters
+ * @param customParams additional input parameters that to add to the ones defined within the API endpoint
+ * @returns
+ */
+const selectCompositeEndpoint = <ConfigDownstream extends Config, ConfigUpstream extends Config>(
+  request: AdapterRequest,
+  downstreamConfig: ConfigDownstream,
+  downstreamEndpoints: Record<string, APIEndpoint<ConfigDownstream>>,
+  upstreamEndpointsGroups: UpstreamEndpointsGroup[],
+  upstreamConfig: ConfigUpstream,
+  ignoreRequired = false,
+  customParams?: InputParameters,
+): APIEndpoint<ConfigDownstream> => {
+  const downstreamEndpoint = selectEndpoint(
+    request,
+    downstreamConfig,
+    downstreamEndpoints,
+    customParams,
+  )
+
+  let additionalInputParameters = {}
+
+  for (const [inputPropertyName, adapterMap, upstreamEndpointName] of upstreamEndpointsGroups) {
+    const keyFromRequest = request.data[inputPropertyName]
+    const upstreamAPIEndpoints = adapterMap[keyFromRequest?.toUpperCase()]
+
+    if (!upstreamAPIEndpoints) {
+      logger.warn(
+        `The request provided a ${inputPropertyName} of ${keyFromRequest} that was not found while merging upstream input parameters. It may not be supported or may be miconfgiured. SKIPPING`,
```

Comment:
```suggestion
        `The request provided a ${inputPropertyName} of ${keyFromRequest} that was not found while merging upstream input parameters. It may not be supported or may be misconfigured. SKIPPING`,
```